packages/manager/src/features/NodeBalancers/NodeBalancerDetail/NodeBalancerDetail.tsx

```diff
@@ -100,7 +101,10 @@ export const NodeBalancerDetail = () => {
             editableTextTitle: nodeBalancerLabel,
             errorText: labelError,
             onCancel: cancelUpdate,
-            onEdit: (label) => updateNodeBalancer({ label }),
+            onEdit: async (label) => {
+              await updateNodeBalancer({ label });
+              refetch();
+            },
```

Comment:
This is not our recommend way of handing cache updates.

```suggestion
            onEdit: async (label) => {
              await updateNodeBalancer({ label });
              refetch();
            },
```

We need to update `packages/queries/src/nodebalancers/nodebalancers.ts` somewhat heavily to account for the conditional call for v4 and v4beta

We need to invalidate the nodebalancer query differently. Let me know if you need more details on what this looks like

It looks like there are now redundant "getNodebalancer" query keys that need to be consolidated